### ISSUE TITLE
License is not changed for beta versions if modified from "Manage Authors & License" page

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-server#4637]
STR:
1. Submit a listed add-on in AMO-stage https://addons.allizom.org/en-US/developers/addon/submit/distribution (approve it)
2. Add a second beta version ("on this site" distribution)
3. Go to "Manage Authors & License" and change add-on license
4. Go to beta version manage page and observe the license 

Expected results:
The new, changed license is displayed.

Actual results:
The old license is displayed.

Notes/Issues:
- the license is changed for normal-listed versions
- if another beta version is added after step 3, its license is ok
Verified on FF51(Win 7). Issue is reproducing on AMO-stage and AMO-dev.
Video for this issue: 
![license1](https://cloud.githubusercontent.com/assets/15685960/22738582/aa7c9ec6-ee10-11e6-9140-91c2d7aaf015.gif)

**Comment by eviljeff** [comment]
I'm tempted to say we shouldn't have the source code license shown in authors & license anyway as it's a version specific setting.  Replace it with a message saying change it in manage status & versions.  Either that or add a heading saying what version's source code license you're actually looking at/changing.
@jvillalobos thoughts?

**Comment by jvillalobos** [comment]
I thought that changing the license in the dev tools didn't do anything for past versions, and just defined how future versions would be licensed. Maybe it only changes the current one and none of the previous ones? That would sort of make sense and make this a wontfix.

I agree it would make more sense to make this a version-specific feature and re-use the license of the previous version by default when submitting a new one. Not a priority, though.

**Comment by eviljeff** [comment]
> I thought that changing the license in the dev tools didn't do anything for past versions, and just defined how future versions would be licensed. Maybe it only changes the current one and none of the previous ones?

It shows/changes the current version.

> That would sort of make sense and make this a wontfix.

Beta, or a version pending review on an approved addon, isn't the current version, though you may think that if that was the last version you uploaded.

> I agree it would make more sense to make this a version-specific feature and re-use the license of the previous version by default when submitting a new one. Not a priority, though.

Good, we already do that all that :)  It _is_ a version specific feature, but we show the source code license on the page with EULA and team members which are addon specific settings.

**Comment by muffinresearch** [comment]
Closing since beta versions are being removed.
